### PR TITLE
Feature #172552417 – Reengineer experiment collections

### PR DESCRIPTION
**Pull request by alfonsomunozpomer**
I pulled the experiment collection logic to `atlas-web-core` but added some logic so that if any of the experiment collections are missing (e.g. bulk) it will fall back to empty optionals and collections. I simplified some of the previous logic:
- I removed the joins for the `SELECT` statement on `experiment2collection` since we map to collection objects by their IDs
- Removed the interface pattern now that all the logic is in `atlas-web-core`

The reason for this change is, mainly, to have everything in `atlas-web-core` and to be able to use the byte array in the icon column of the `collections` table.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/atlas/experiments/collections/ExperimentCollection.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        try (var imageByteArrayInputStream = new ByteArrayInputStream(imageBytes)) {
+            return new AutoValue_ExperimentCollection(
+                    id,
+                    name,
+                    description,
+                    Optional.of(ImageIO.read(imageByteArrayInputStream)));
+        } catch (Exception e) {
+            // Maybe too defensive?
+            return new AutoValue_ExperimentCollection(id, name, description, Optional.empty());
```
Comment (haideriqbal) [inline src/main/java/uk/ac/ebi/atlas/experiments/collections/ExperimentCollection.java:36]: 👍

File: src/main/java/uk/ac/ebi/atlas/experiments/collections/ExperimentCollectionDao.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+@Repository
+@Transactional(transactionManager = "txManager", readOnly = true)
```
Comment (haideriqbal) [inline src/main/java/uk/ac/ebi/atlas/experiments/collections/ExperimentCollectionDao.java:15]: Just a question: I assume its a good practice to wrap up db access code in "Transactional", but is there any specific reason we don't have such "Transactional" annotation for some db related dao classes e.g. `ExperimentTraderDao`

File: src/test/resources/fixtures/experiment2collection-fixture.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO experiment2collection (exp_acc, coll_id) VALUES ('E-MTAB-5061', 'CZB');
+--  No experiment2collection-delete.sql: table deleted in ordered test suite in the last test
```
Comment (haideriqbal) [inline src/test/resources/fixtures/experiment2collection-fixture.sql:2]: 👍

**Review by haideriqbal — APPROVED** [review]
Just one minor comment. Logic seems pretty clean and easy to follow. Also really liked the part of removing joins from db query and splitting the logic into two daos. LGTM!

File: src/main/java/uk/ac/ebi/atlas/experiments/collections/ExperimentCollectionDao.java (hunk 2 above)
Comment (alfonsomunozpomer) [reply src/main/java/uk/ac/ebi/atlas/experiments/collections/ExperimentCollectionDao.java:15]: I guess we forgot or we didn’t know better. :disappointed: In general it’s good practice to do it so if anything goes wrong it’s easier to recover from an exception.

**Pull request merged**